logging/nxscope/nxscope_chan.c

```diff
@@ -0,0 +1,1566 @@
+      goto errout;
+    }
+#endif
+
+#ifdef CONFIG_LOGGING_NXSCOPE_DIVIDER
+  /* Handle sample rate divider */
+
+  s->cntr[ch] += 1;
+  if (s->cntr[ch] % (s->chinfo[ch].div + 1) != 0)
+    {
+      ret = -EAGAIN;
+      goto errout;
+    }
+#endif
+
+  /* Get utype */
+
+  utype.u8 = type;
+
+#ifdef CONFIG_LOGGING_NXSCOPE_CRICHANNELS
+  /* No more checks needed for critical channel */
+
+  if (utype.s.cri)
+    {
+      ret = OK;
+      goto errout;
+    }
+#endif
+
+  /* Check buffer size */
+
+#ifdef CONFIG_LOGGING_NXSCOPE_USERTYPES
+  if (type >= NXSCOPE_TYPE_USER)
+    {
+      type_size = 1;
+    }
+  else
+#endif
+    {
+      type_size = g_type_size[utype.s.dtype];
+    }
+
+  next_i = (s->stream_i + 1 + type_size * d + mlen +
+            s->proto_stream->footlen);
+
+  if (next_i > s->streambuf_len)
+    {
+      _err("ERROR: no space for data %d\n", s->stream_i);
+      nxscope_stream_overflow(s);
+      ret = -ENOBUFS;
+      goto errout;
+    }
+
+errout:
+  return ret;
+}
+
+/****************************************************************************
+ * Name: nxscope_put_vector
+ *
+ * NOTE: This function assumes that we have exclusive access to the nxscope
+ *       stream buffer
+ *
+ * IMPORTANT: Data stored always as little-endian !
+ *
+ ****************************************************************************/
+
+static int nxscope_put_vector(FAR uint8_t *buff, uint8_t type, FAR void *val,
+                              uint8_t d)
+{
+  int i = 0;
+  int j = 0;
+
+  DEBUGASSERT(buff);
+
+  /* Pack data */
+
+  switch (type)
+    {
+      case NXSCOPE_TYPE_NONE:
+        {
+          /* Nothing to do here */
+
+          break;
+        }
+
+      case NXSCOPE_TYPE_UINT8:
+      case NXSCOPE_TYPE_INT8:
+#ifdef CONFIG_LOGGING_NXSCOPE_USERTYPES
+      case NXSCOPE_TYPE_USER:
+#endif
+        {
+          uint8_t u8 = 0;
+
+          for (i = 0; i < d; i++)
+            {
+              DEBUGASSERT(val);
+              u8 = ((FAR uint8_t *)val)[i];
+
+              buff[j++] = u8;
+            }
+
+          break;
+        }
+
+      case NXSCOPE_TYPE_UINT16:
+      case NXSCOPE_TYPE_INT16:
+      case NXSCOPE_TYPE_B8:
+      case NXSCOPE_TYPE_UB8:
+        {
+          uint16_t u16 = 0;
+
+          for (i = 0; i < d; i++)
+            {
+              DEBUGASSERT(val);
+              u16 = ((FAR uint16_t *)val)[i];
+#ifdef CONFIG_ENDIAN_BIG
+              u16 = __swap_uint16(u16);
+#endif
+
+              buff[j++] = ((u16 >> 0) & 0xff);
+              buff[j++] = ((u16 >> 8) & 0xff);
+            }
+
+          break;
+        }
+
+      case NXSCOPE_TYPE_UINT32:
+      case NXSCOPE_TYPE_INT32:
+      case NXSCOPE_TYPE_FLOAT:
+      case NXSCOPE_TYPE_B16:
+      case NXSCOPE_TYPE_UB16:
+        {
+          uint32_t u32 = 0;
+
+          for (i = 0; i < d; i++)
+            {
+              DEBUGASSERT(val);
+              u32 = ((FAR uint32_t *)val)[i];
+#ifdef CONFIG_ENDIAN_BIG
+              u32 = __swap_uint32(u32);
```

Comment:
How about using `htole32()` and family from `endian.h` file?

Reply:
good idea. done